google/cloud/aiplatform/metadata/experiment_resources.py

```diff
@@ -0,0 +1,717 @@
+                project=project,
+                location=location,
+                credentials=credentials,
+            )
+
+        self = cls.__new__()
+        self._metadata_context = experiment_context
+
+        return self
+
+    @classmethod
+    def get_or_create(
+        cls,
+        experiment_name: str,
+        *,
+        description: Optional[str] = None,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> "Experiment":
+        """Get's experiment if one exists with this in Vertex AI Experiments. Otherwise creates this experiment.
+
+        ```
+        my_experiment = aiplatform.Experiment.get_or_create('my-experiment', description='my description')
+        ```
+
+        Args:
+            experiment_name (str): Required. The name of this experiment.
+            description (str): Optional. Describes this experiment's purpose.
+            project (str):
+                Optional. Project where this experiment will be retrieved from or created. Overrides project set in
+                aiplatform.init.
+            location (str):
+                Optional. Location where this experiment will be retrieved from or created. Overrides location set in
+                aiplatform.init.
+            credentials (auth_credentials.Credentials):
+                Optional. Custom credentials used to retrieve or create this experiment. Overrides
+                credentials set in aiplatform.init.
+        Returns:
+            Vertex AI experiment.
+        """
+
+        metadata_store._MetadataStore.ensure_default_metadata_store_exists(
+            project=project, location=location, credentials=credentials
+        )
+
+        with _SetLoggerLevel(resource):
+            experiment_context = context._Context.get_or_create(
+                resource_id=experiment_name,
+                display_name=experiment_name,
+                description=description,
+                schema_title=constants.SYSTEM_EXPERIMENT,
+                schema_version=metadata._get_experiment_schema_version(),
+                metadata=constants.EXPERIMENT_METADATA,
+                project=project,
+                location=location,
+                credentials=credentials,
+            )
+
+        cls._validate_experiment_context(experiment_context)
+
+        if description and description != experiment_context.description:
+            experiment_context.update(description=description)
+
+        self = cls.__new__(cls)
+        self._metadata_context = experiment_context
+
+        return self
+
+    @classmethod
+    def list(
+        cls,
+        *,
+        project: Optional[str] = None,
+        location: Optional[str] = None,
+        credentials: Optional[auth_credentials.Credentials] = None,
+    ) -> List["Experiment"]:
+        """List experiments in Vertex AI Experiments.
```

Comment:
For clarity, consider changing to "List all Vertex AI Experiments in the given project"

Reply:
done